tmt/base.py

```diff
@@ -2593,13 +2593,18 @@ def __init__(self, data, name=None, test=None):
         self.note = data.get('note')
         self.duration = data.get('duration')
         if test:
-            self.id = test.node.get(
-                'id', test.node.get(
-                    'extra-nitrate', test.node.get(
-                        'extra-task', '')))
+            # Saving identifiable information for each test case so we can match them
+            # to Polarion/Nitrate/other cases and report run results there
+            self.ids = {
+                tmt.identifier.ID_KEY: test.node.get(tmt.identifier.ID_KEY),
+                'extra-nitrate': test.node.get('extra-nitrate'),
+                'extra-task': test.node.get('extra-task')}
```

Comment:
* `going after test ID directly into fmf node shouldn't be the first choice, `test.id` is the correct property - validated and normalized.
* extracting `extra-*` keys - and just a very limited set of all possible and allowed `extra-*` keys - seems like something `Result` should support in a slightly different fashion. At least these two keys should be extracted into a global list, like `TEST_KEYS_WE_TREAT_AS_IDS_AND_PROPAGATE_TO_RESULTS`. One day, I might want to add a custom `extra-my-awesome-tcms-clone-id` key to `result.ids`, and if `extra-*` are supported, there should be a way to extend this behavior for new keys people add.

(Preferably, `base.py` should never ever be bothered with bits like `nitrate`, there's no point in base classes being aware of the existence of such a specific system as nitrate or polarion - yes, "expose test IDs, propagate these keys into result IDs" as a concept, absolutely, but why should `Result` be aware of one of many possible ID namespaces... Hardcoding `extra-polarion`, `extra-jira`, `extra-zephyr` keys into this very place doesn't sound like a good idea - it should be way more open and generic, for plugins to use).